File: android/app/src/reminder/java/com/meulembrete/app/MainActivity.java
package com.meulembrete.app;

import android.Manifest;
import android.app.Activity;
import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.content.pm.PackageManager;
import android.graphics.Color;
import android.graphics.Typeface;
import android.graphics.drawable.GradientDrawable;
import android.os.Build;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.temporal.ChronoUnit;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class MainActivity extends Activity {
    private static final int GREEN = Color.rgb(22, 138, 85);
    private static final int BG = Color.rgb(245, 248, 246);
    private static final int SURFACE = Color.WHITE;
    private static final int SOFT_GREEN = Color.rgb(232, 245, 238);
    private static final int TEXT = Color.rgb(31, 41, 36);
    private static final int MUTED = Color.rgb(94, 109, 101);
    private static final String[] CATEGORIES = {"💼 Trabalho", "❤️ Saúde", "💰 Financeiro", "🏠 Pessoal", "👨‍👩‍👧 Família", "🎉 Evento", "📚 Estudos", "📌 Outro"};

    private ReminderStore store;
    private List<Commitment> commitments = new ArrayList<>();
    private LinearLayout listContainer;
    private TextView dailySummary, todayStat, weekStat, activeStat;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        store = new ReminderStore(this);
        ReminderScheduler.scheduleDaily(this);
        requestNotificationPermission();
        buildUi();
        refresh();
    }

    private void requestNotificationPermission() {
        if (Build.VERSION.SDK_INT >= 33 && checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
            requestPermissions(new String[]{Manifest.permission.POST_NOTIFICATIONS}, 1001);
        }
    }

    private void buildUi() {
        getWindow().setStatusBarColor(BG);
        getWindow().setNavigationBarColor(BG);
        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setBackgroundColor(BG);

        ScrollView scroll = new ScrollView(this);
        scroll.setFillViewport(true);
        root.addView(scroll, new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 0, 1f));

        LinearLayout content = new LinearLayout(this);
        content.setOrientation(LinearLayout.VERTICAL);
        content.setPadding(dp(16), dp(18), dp(16), dp(22));
        scroll.addView(content, new ScrollView.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        content.addView(text("Meu Lembrete", 24, TEXT, Typeface.BOLD));
        content.addView(text("Seus compromissos sem complicação", 14, MUTED, Typeface.NORMAL), withTop(dp(2)));

        LinearLayout header = panel(SOFT_GREEN, 26);
        header.setPadding(dp(18), dp(18), dp(18), dp(18));
        LinearLayout.LayoutParams hp = matchWrap(); hp.topMargin = dp(18); content.addView(header, hp);
        header.addView(text("🔔  RESUMO DIÁRIO", 13, GREEN, Typeface.BOLD));
        dailySummary = text("", 18, TEXT, Typeface.BOLD); header.addView(dailySummary, withTop(dp(9)));
        header.addView(text("Todos os dias o app informa o compromisso, a categoria e quantos dias faltam.", 14, MUTED, Typeface.NORMAL), withTop(dp(8)));

        LinearLayout stats = new LinearLayout(this); stats.setOrientation(LinearLayout.HORIZONTAL);
        LinearLayout.LayoutParams sp = matchWrap(); sp.topMargin = dp(12); content.addView(stats, sp);
        todayStat = addStat(stats, "Hoje"); weekStat = addStat(stats, "7 dias"); activeStat = addStat(stats, "Ativos");

        Button test = primaryButton("🔔  Testar notificação agora", true);
        test.setOnClickListener(v -> testNotification());
        LinearLayout.LayoutParams tp = matchWrap(); tp.topMargin = dp(14); content.addView(test, tp);

        LinearLayout.LayoutParams stp = wrapWrap(); stp.topMargin = dp(24);
        content.addView(text("Próximos compromissos", 20, TEXT, Typeface.BOLD), stp);
        content.addView(text("Os mais próximos aparecem primeiro", 14, MUTED, Typeface.NORMAL), withTop(dp(3)));
        listContainer = new LinearLayout(this); listContainer.setOrientation(LinearLayout.VERTICAL); content.addView(listContainer, withTop(dp(10)));

        Button add = primaryButton("＋  Adicionar compromisso", false); add.setOnClickListener(v -> showAddDialog());
        LinearLayout.LayoutParams ap = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, dp(58)); ap.setMargins(dp(16), dp(10), dp(16), dp(16)); root.addView(add, ap);
        setContentView(root);
    }

    private TextView addStat(LinearLayout parent, String label) {
        LinearLayout box = panel(SURFACE, 20); box.setGravity(Gravity.CENTER_HORIZONTAL); box.setPadding(dp(8), dp(12), dp(8), dp(12));
        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1f); p.setMargins(dp(3), 0, dp(3), 0); parent.addView(box, p);
        TextView value = text("0", 23, TEXT, Typeface.BOLD); value.setGravity(Gravity.CENTER); box.addView(value);
        TextView lbl = text(label, 12, MUTED, Typeface.NORMAL); lbl.setGravity(Gravity.CENTER); box.addView(lbl, withTop(dp(2))); return value;
    }

    private void refresh() {
        commitments = store.getAll(); commitments.sort(Comparator.comparing(c -> c.date == null ? "" : c.date));
        LocalDate today = LocalDate.now(); int active = 0, todayCount = 0, weekCount = 0; Commitment nearest = null; long nearestDays = Long.MAX_VALUE;
        for (Commitment item : commitments) {
            if (item.completed) continue; active++; Long days = daysUntil(item.date, today); if (days == null) continue;
            if (days == 0) todayCount++; if (days >= 0 && days <= 7) weekCount++;
            if (days >= 0 && days < nearestDays) { nearestDays = days; nearest = item; }
        }
        todayStat.setText(String.valueOf(todayCount)); weekStat.setText(String.valueOf(weekCount)); activeStat.setText(String.valueOf(active));
        dailySummary.setText(nearest == null ? "Nenhum compromisso futuro pendente." : nearest.category + "  " + nearest.title + " • " + remaining(nearestDays));
        listContainer.removeAllViews();
        if (commitments.isEmpty()) {
            LinearLayout empty = panel(SURFACE, 22); empty.setPadding(dp(18), dp(20), dp(18), dp(20));
            empty.addView(text("Você ainda não cadastrou compromissos.", 16, TEXT, Typeface.BOLD));
            empty.addView(text("Toque em “Adicionar compromisso” para começar.", 14, MUTED, Typeface.NORMAL), withTop(dp(6))); listContainer.addView(empty, matchWrap()); return;
        }
        for (Commitment item : commitments) listContainer.addView(commitmentCard(item), cardParams());
    }

    private View commitmentCard(Commitment item) {
        LinearLayout card = panel(SURFACE, 22); card.setPadding(dp(16), dp(16), dp(16), dp(14));
        LinearLayout top = new LinearLayout(this); top.setOrientation(LinearLayout.HORIZONTAL); top.setGravity(Gravity.TOP); card.addView(top, matchWrap());
        LinearLayout left = new LinearLayout(this); left.setOrientation(LinearLayout.VERTICAL); top.addView(left, new LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1f));
        left.addView(text(item.category, 13, GREEN, Typeface.BOLD));
        TextView title = text(item.title, 18, TEXT, Typeface.BOLD); if (item.completed) title.setAlpha(0.5f); left.addView(title, withTop(dp(3)));
        String when = formatDate(item.date) + (item.time == null || item.time.trim().isEmpty() ? "" : " • " + item.time.trim()); left.addView(text(when, 13, MUTED, Typeface.NORMAL), withTop(dp(3)));
        Long days = daysUntil(item.date, LocalDate.now()); String badgeText = item.completed ? "Concluído" : (days == null ? "Data" : remaining(days));
        TextView badge = text(badgeText, 12, item.completed ? MUTED : GREEN, Typeface.BOLD); badge.setGravity(Gravity.CENTER); badge.setPadding(dp(10), dp(7), dp(10), dp(7)); badge.setBackground(round(SOFT_GREEN, 50));
        LinearLayout.LayoutParams bp = wrapWrap(); bp.leftMargin = dp(8); top.addView(badge, bp);
        if (item.notes != null && !item.notes.trim().isEmpty()) card.addView(text(item.notes.trim(), 14, MUTED, Typeface.NORMAL), withTop(dp(11)));
        LinearLayout actions = new LinearLayout(this); actions.setOrientation(LinearLayout.HORIZONTAL); LinearLayout.LayoutParams acp = matchWrap(); acp.topMargin = dp(12); card.addView(actions, acp);
        Button toggle = secondaryButton(item.completed ? "Reabrir" : "✓ Concluir"); toggle.setOnClickListener(v -> { item.completed = !item.completed; store.saveAll(commitments); refresh(); }); actions.addView(toggle, new LinearLayout.LayoutParams(0, dp(44), 1f));
        Button delete = secondaryButton("Excluir"); delete.setTextColor(Color.rgb(166, 48, 48)); delete.setOnClickListener(v -> { commitments.removeIf(c -> c.id == item.id); store.saveAll(commitments); refresh(); });
        LinearLayout.LayoutParams dp = new LinearLayout.LayoutParams(0, this.dp(44), 1f); dp.leftMargin = this.dp(8); actions.addView(delete, dp); return card;
    }

    private void testNotification() {
        if (Build.VERSION.SDK_INT >= 33 && checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) { requestNotificationPermission(); Toast.makeText(this, "Permita as notificações e toque novamente para testar.", Toast.LENGTH_LONG).show(); return; }
        boolean hasActive = false; for (Commitment c : commitments) if (!c.completed) { hasActive = true; break; }
        if (!hasActive) { Toast.makeText(this, "Cadastre um compromisso primeiro para testar o aviso.", Toast.LENGTH_LONG).show(); return; }
        ReminderScheduler.testNow(this); Toast.makeText(this, "Notificação de teste enviada.", Toast.LENGTH_SHORT).show();
    }

    private void showAddDialog() {
        LinearLayout form = new LinearLayout(this); form.setOrientation(LinearLayout.VERTICAL); form.setPadding(dp(22), dp(8), dp(22), dp(4));
        EditText title = new EditText(this); title.setHint("Nome do compromisso"); title.setSingleLine(true); form.addView(title, matchWrap());
        Spinner category = new Spinner(this); category.setAdapter(new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, CATEGORIES)); form.addView(category, withTop(dp(10)));
        final LocalDate[] selectedDate = {LocalDate.now().plusDays(1)};
        Button dateButton = secondaryButton("Data: " + formatDate(selectedDate[0].toString()));
        dateButton.setOnClickListener(v -> { LocalDate current = selectedDate[0]; new DatePickerDialog(this, (view, year, month, day) -> { selectedDate[0] = LocalDate.of(year, month + 1, day); dateButton.setText("Data: " + formatDate(selectedDate[0].toString())); }, current.getYear(), current.getMonthValue() - 1, current.getDayOfMonth()).show(); });
        LinearLayout.LayoutParams dateParams = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, dp(50)); dateParams.topMargin = dp(10); form.addView(dateButton, dateParams);
        EditText time = new EditText(this); time.setHint("Horário opcional (ex.: 14:30)"); time.setSingleLine(true); form.addView(time, withTop(dp(8)));
        EditText notes = new EditText(this); notes.setHint("Observações: documentos, endereço, detalhes..."); notes.setMinLines(2); form.addView(notes, withTop(dp(8)));
        AlertDialog dialog = new AlertDialog.Builder(this).setTitle("Novo compromisso").setView(form).setNegativeButton("Cancelar", null).setPositiveButton("Salvar", null).create();
        dialog.setOnShowListener(d -> dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(v -> {
            String name = title.getText().toString().trim(); if (name.isEmpty()) { title.setError("Informe o nome do compromisso"); return; }
            commitments.add(new Commitment(System.currentTimeMillis(), name, category.getSelectedItem().toString(), selectedDate[0].toString(), time.getText().toString().trim(), notes.getText().toString().trim(), false));
            store.saveAll(commitments); ReminderScheduler.scheduleDaily(this); dialog.dismiss(); refresh(); Toast.makeText(this, "Compromisso salvo.", Toast.LENGTH_SHORT).show();
        })); dialog.show();
    }

    private Long daysUntil(String isoDate, LocalDate today) { try { return ChronoUnit.DAYS.between(today, LocalDate.parse(isoDate)); } catch (Exception e) { return null; } }
    private String remaining(long days) { if (days < 0) return "atrasado há " + Math.abs(days) + (Math.abs(days) == 1 ? " dia" : " dias"); if (days == 0) return "é hoje"; if (days == 1) return "falta 1 dia"; return "faltam " + days + " dias"; }
    private String formatDate(String iso) { try { return LocalDate.parse(iso).format(DateTimeFormatter.ofPattern("dd/MM/yyyy")); } catch (Exception e) { return iso; } }
    private LinearLayout panel(int color, int radiusDp) { LinearLayout layout = new LinearLayout(this); layout.setOrientation(LinearLayout.VERTICAL); layout.setBackground(round(color, radiusDp)); return layout; }
    private GradientDrawable round(int color, int radiusDp) { GradientDrawable d = new GradientDrawable(); d.setColor(color); d.setCornerRadius(dp(radiusDp)); return d; }
    private TextView text(String value, float size, int color, int style) { TextView t = new TextView(this); t.setText(value); t.setTextSize(size); t.setTextColor(color); t.setTypeface(Typeface.create(Typeface.DEFAULT, style)); t.setLineSpacing(0, 1.08f); return t; }
    private Button primaryButton(String label, boolean soft) { Button b = new Button(this); b.setText(label); b.setTextSize(15); b.setTypeface(Typeface.DEFAULT, Typeface.BOLD); b.setAllCaps(false); b.setTextColor(soft ? GREEN : Color.WHITE); b.setBackground(round(soft ? SOFT_GREEN : GREEN, 18)); return b; }
    private Button secondaryButton(String label) { Button b = new Button(this); b.setText(label); b.setTextSize(14); b.setAllCaps(false); b.setTextColor(GREEN); GradientDrawable d = round(Color.WHITE, 15); d.setStroke(dp(1), Color.rgb(220, 228, 223)); b.setBackground(d); return b; }
    private LinearLayout.LayoutParams matchWrap() { return new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT); }
    private LinearLayout.LayoutParams wrapWrap() { return new LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT); }
    private LinearLayout.LayoutParams withTop(int top) { LinearLayout.LayoutParams p = matchWrap(); p.topMargin = top; return p; }
    private LinearLayout.LayoutParams cardParams() { LinearLayout.LayoutParams p = matchWrap(); p.bottomMargin = dp(10); return p; }
    private int dp(int value) { return Math.round(value * getResources().getDisplayMetrics().density); }
}

File: meu-lembrete-android/app/src/main/java/com/meulembrete/app/MainActivity.kt
package com.meulembrete.app

import android.Manifest
import android.app.DatePickerDialog
import android.os.Build
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.time.temporal.ChronoUnit

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ReminderScheduler.schedule(this)
        setContent { MeuLembreteApp() }
    }
}

private val categories = listOf(
    "💼 Trabalho", "❤️ Saúde", "💰 Financeiro", "🏠 Pessoal",
    "👨‍👩‍👧 Família", "🎉 Evento", "📚 Estudos", "📌 Outro"
)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MeuLembreteApp() {
    val context = LocalContext.current
    val repository = remember { CommitmentRepository(context) }
    var commitments by remember { mutableStateOf(repository.getAll()) }
    var showAdd by remember { mutableStateOf(false) }

    val permissionLauncher = rememberLauncherForActivityResult(
        ActivityResultContracts.RequestPermission()
    ) { }

    LaunchedEffect(Unit) {
        if (Build.VERSION.SDK_INT >= 33) {
            permissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
        }
    }

    MaterialTheme(
        colorScheme = lightColorScheme(
            primary = androidx.compose.ui.graphics.Color(0xFF168A55),
            secondary = androidx.compose.ui.graphics.Color(0xFF3F765B),
            surface = androidx.compose.ui.graphics.Color(0xFFF9FCFA),
            background = androidx.compose.ui.graphics.Color(0xFFF5F8F6)
        )
    ) {
        Scaffold(
            containerColor = MaterialTheme.colorScheme.background,
            floatingActionButton = {
                FloatingActionButton(onClick = { showAdd = true }) {
                    Icon(Icons.Default.Add, contentDescription = "Adicionar compromisso")
                }
            }
        ) { padding ->
            LazyColumn(
                modifier = Modifier.fillMaxSize().padding(padding),
                contentPadding = PaddingValues(16.dp, 18.dp, 16.dp, 100.dp),
                verticalArrangement = Arrangement.spacedBy(14.dp)
            ) {
                item { HeaderCard(commitments) }
                item { SummaryCards(commitments) }
                item {
                    FilledTonalButton(
                        onClick = { ReminderScheduler.testNow(context) },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Icon(Icons.Default.PlayArrow, contentDescription = null)
                        Spacer(Modifier.width(8.dp))
                        Text("Testar notificação agora")
                    }
                }
                item {
                    Text("Próximos compromissos", style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.SemiBold)
                    Text("Os mais próximos aparecem primeiro", color = MaterialTheme.colorScheme.onSurfaceVariant)
                }

                val ordered = commitments.sortedBy { it.date }
                if (ordered.isEmpty()) {
                    item {
                        Card(Modifier.fillMaxWidth(), shape = RoundedCornerShape(24.dp)) {
                            Column(Modifier.padding(22.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                                Icon(Icons.Default.Notifications, null, tint = MaterialTheme.colorScheme.primary)
                                Text("Nenhum compromisso cadastrado", fontWeight = FontWeight.SemiBold)
                                Text("Toque no + para adicionar o primeiro lembrete.")
                            }
                        }
                    }
                } else {
                    items(ordered, key = { it.id }) { item ->
                        CommitmentCard(
                            item = item,
                            onToggle = {
                                commitments = commitments.map { if (it.id == item.id) it.copy(completed = !it.completed) else it }
                                repository.saveAll(commitments)
                            },
                            onDelete = {
                                commitments = commitments.filterNot { it.id == item.id }
                                repository.saveAll(commitments)
                            }
                        )
                    }
                }
            }
        }

        if (showAdd) {
            AddCommitmentDialog(
                onDismiss = { showAdd = false },
                onSave = { newItem ->
                    commitments = commitments + newItem
                    repository.saveAll(commitments)
                    ReminderScheduler.schedule(context)
                    showAdd = false
                }
            )
        }
    }
}

@Composable
private fun HeaderCard(items: List<Commitment>) {
    val upcoming = items.filter { !it.completed }.mapNotNull { item ->
        runCatching { item to ChronoUnit.DAYS.between(LocalDate.now(), LocalDate.parse(item.date)) }.getOrNull()
    }.filter { it.second >= 0 }.minByOrNull { it.second }

    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(28.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primaryContainer)
    ) {
        Column(Modifier.padding(22.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Icon(Icons.Default.Notifications, null, tint = MaterialTheme.colorScheme.primary)
                Text("Resumo diário", fontWeight = FontWeight.SemiBold)
            }
            Text(
                upcoming?.let { (item, days) ->
                    when (days) {
                        0L -> "${item.category}  ${item.title} é hoje"
                        1L -> "${item.category}  ${item.title}: falta 1 dia"
                        else -> "${item.category}  ${item.title}: faltam $days dias"
                    }
                } ?: "Você não tem compromissos futuros pendentes.",
                style = MaterialTheme.typography.titleMedium
            )
            Text("O app prepara um aviso diário com nome, categoria e contagem regressiva.")
        }
    }
}

@Composable
private fun SummaryCards(items: List<Commitment>) {
    val active = items.filter { !it.completed }
    val today = LocalDate.now()
    fun countUntil(max: Long) = active.count {
        runCatching { ChronoUnit.DAYS.between(today, LocalDate.parse(it.date)) }.getOrNull()?.let { d -> d in 0..max } == true
    }
    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
        MiniStat("Hoje", countUntil(0), Modifier.weight(1f))
        MiniStat("7 dias", countUntil(7), Modifier.weight(1f))
        MiniStat("Ativos", active.size, Modifier.weight(1f))
    }
}

@Composable
private fun MiniStat(label: String, value: Int, modifier: Modifier = Modifier) {
    Card(modifier, shape = RoundedCornerShape(20.dp)) {
        Column(Modifier.padding(14.dp)) {
            Text(label, style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurfaceVariant)
            Text(value.toString(), style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.Bold)
        }
    }
}

@Composable
private fun CommitmentCard(item: Commitment, onToggle: () -> Unit, onDelete: () -> Unit) {
    val days = runCatching { ChronoUnit.DAYS.between(LocalDate.now(), LocalDate.parse(item.date)) }.getOrNull()
    val countdown = when {
        item.completed -> "Concluído"
        days == null -> "Data inválida"
        days < 0 -> "Atrasado"
        days == 0L -> "É hoje"
        days == 1L -> "Falta 1 dia"
        else -> "Faltam $days dias"
    }

    Card(Modifier.fillMaxWidth(), shape = RoundedCornerShape(24.dp)) {
        Column(Modifier.padding(18.dp), verticalArrangement = Arrangement.spacedBy(12.dp)) {
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.Top) {
                Column(Modifier.weight(1f)) {
                    Text(item.category, style = MaterialTheme.typography.labelLarge, color = MaterialTheme.colorScheme.primary)
                    Text(item.title, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
                    val formatted = runCatching { LocalDate.parse(item.date).format(DateTimeFormatter.ofPattern("dd/MM/yyyy")) }.getOrDefault(item.date)
                    Text("$formatted${if (item.time.isNotBlank()) " • ${item.time}" else ""}", color = MaterialTheme.colorScheme.onSurfaceVariant)
                }
                Surface(shape = RoundedCornerShape(50), color = MaterialTheme.colorScheme.primaryContainer) {
                    Text(countdown, Modifier.padding(horizontal = 12.dp, vertical = 7.dp), fontWeight = FontWeight.Medium)
                }
            }
            if (item.notes.isNotBlank()) Text(item.notes, style = MaterialTheme.typography.bodyMedium)
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                FilledTonalButton(onClick = onToggle) {
                    Icon(Icons.Default.Check, null)
                    Spacer(Modifier.width(6.dp))
                    Text(if (item.completed) "Reabrir" else "Concluir")
                }
                TextButton(onClick = onDelete) {
                    Icon(Icons.Default.Delete, null)
                    Spacer(Modifier.width(4.dp))
                    Text("Excluir")
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun AddCommitmentDialog(onDismiss: () -> Unit, onSave: (Commitment) -> Unit) {
    val context = LocalContext.current
    var title by remember { mutableStateOf("") }
    var category by remember { mutableStateOf(categories.first()) }
    var date by remember { mutableStateOf(LocalDate.now().plusDays(1).toString()) }
    var time by remember { mutableStateOf("") }
    var notes by remember { mutableStateOf("") }
    var expanded by remember { mutableStateOf(false) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Novo compromisso") },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                OutlinedTextField(title, { title = it }, label = { Text("Nome do compromisso") }, singleLine = true, modifier = Modifier.fillMaxWidth())
                ExposedDropdownMenuBox(expanded = expanded, onExpandedChange = { expanded = !expanded }) {
                    OutlinedTextField(
                        value = category,
                        onValueChange = {},
                        readOnly = true,
                        label = { Text("Categoria") },
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded) },
                        modifier = Modifier.menuAnchor().fillMaxWidth()
                    )
                    ExposedDropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
                        categories.forEach { option ->
                            DropdownMenuItem(text = { Text(option) }, onClick = { category = option; expanded = false })
                        }
                    }
                }
                OutlinedButton(
                    onClick = {
                        val current = LocalDate.parse(date)
                        DatePickerDialog(
                            context,
                            { _, y, m, d -> date = LocalDate.of(y, m + 1, d).toString() },
                            current.year, current.monthValue - 1, current.dayOfMonth
                        ).show()
                    },
                    modifier = Modifier.fillMaxWidth()
                ) { Text("Data: ${LocalDate.parse(date).format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}") }
                OutlinedTextField(time, { time = it }, label = { Text("Horário (opcional, ex.: 14:30)") }, singleLine = true, modifier = Modifier.fillMaxWidth())
                OutlinedTextField(notes, { notes = it }, label = { Text("Observações") }, minLines = 2, modifier = Modifier.fillMaxWidth())
            }
        },
        confirmButton = {
            Button(
                enabled = title.isNotBlank(),
                onClick = {
                    onSave(
                        Commitment(
                            id = System.currentTimeMillis(),
                            title = title.trim(),
                            category = category,
                            date = date,
                            time = time.trim(),
                            notes = notes.trim()
                        )
                    )
                }
            ) { Text("Salvar") }
        },
        dismissButton = { TextButton(onClick = onDismiss) { Text("Cancelar") } }
    )
}
